curriculum/challenges/english/07-scientific-computing-with-python/scientific-computing-with-python-projects/probability-calculator.md

```diff
@@ -140,6 +143,49 @@ t.result.wasSuccessful()
 });
 ```
 
+The `draw` method should behave correctly when the number of balls to extract is bigger than the number of balls in the hat.
+
+```js
+({
+  test: () => {
+    pyodide.FS.writeFile("/home/pyodide/probability_calculator.py", code);
+    pyodide.FS.writeFile(
+      "/home/pyodide/test_module.py",
+      `
+import unittest
+import probability_calculator
+from importlib import reload
+
+reload(probability_calculator)
+
+probability_calculator.random.seed(95)
+class UnitTests(unittest.TestCase):
+    maxDiff = None
+    def test_hat_draw_2(self):
+        hat = probability_calculator.Hat(yellow=5,red=1,green=3,blue=9,test=1)
+        actual = hat.draw(20)
+        expected = ['yellow', 'yellow', 'yellow', 'yellow', 'yellow', 'red', 'green', 'green', 'green', 'blue', 'blue', 'blue', 'blue', 'blue', 'blue', 'blue', 'blue', 'blue', 'test']
+        self.assertEqual(actual, expected, 'Expected hat draw to return all items from hat contents.')
+        actual = len(hat.contents)
+        expected = 0
+        self.assertEqual(actual, expected, 'Expected hat draw to leave no items in contents.')
```

Comment:
To be honest I am not sure about this part: should the hat be emptied extracting a number of balls greater than the number of balls in the hat?

Reply:
Yes.